cni/pkg/cmd/root.go

```diff
@@ -99,11 +99,12 @@ var rootCmd = &cobra.Command{
 			log.Infof("Starting ambient node agent with inpod redirect mode on socket %s", cniEventAddr)
 			ambientAgent, err := nodeagent.NewServer(ctx, watchServerReady, cniEventAddr,
 				nodeagent.AmbientArgs{
```

Comment:
Not sure where in the PR this is caused, so the line I am commenting is random./

With this change, when we setup host iptables rules we are dumping the entire host ruleset to logs. This can be **massive** and probably not something we want.

Note sure we even want to run `iptables-save` at all there (unless we already are)

Reply:
Also logs:

```
[pod/istio-cni-node-xflzt/install-cni] 2024-11-18T19:27:07.777340Z      warn    cni-agent       Found residues of old iptables rules/chains, reconciliation is needed   component=host
[pod/istio-cni-node-xflzt/install-cni] 2024-11-18T19:27:07.777344Z      warn    cni-agent       reconcile is needed but no-reconcile flag is set. Unexpected behavior may occur due to preexist
ing iptables rules      component=host
```

IIUC this is entirely normal behavior. Normal operations should not `warn`

Reply:
I agree that dumping the iptables-save output is excessive for the host. I will move it to the debug level,  I cannot completely eliminate it since that information is invaluable for troubleshooting when needed.

> IUC this is entirely normal behavior. Normal operations should not warn

The log level was inherited from Istio-init idempotency, where any "residue" of old iptables is indeed something to be warned about. Now that the implementation is shared, I will reduce the log level.

> Note sure we even want to run iptables-save at all there (unless we already are)

The `iptables-save` cmd on the host is used to verify that compatible rules are not already present. If compatible rules are found, they are skipped. However, since we perform a delete-and-apply operation when the server starts, these compatible rules will never be found, making the check somewhat redundant.

The issue is that to skip this check, I would need to add a new iptables flag to completely disable idempotency, which would remove the need for iptables-save altogether. Considering that it would be beneficial to enable reconciliation for the host in the future, I feel it may not be worth the effort to make this change now.

Reply:
iptables-save is now silenced and the log level of those messages has been changed